parity/configuration.rs

```diff
@@ -468,8 +471,9 @@ impl Configuration {
 	}
 
 	fn max_peers(&self) -> u32 {
-		let peers = self.args.arg_max_peers as u32;
-		max(self.min_peers(), peers)
+		self.args.arg_max_peers
+			.or(max(self.args.arg_min_peers, Some(DEFAULT_MAX_PEERS)))
```

Comment:
This will always return `T` as long at least one of the arguments is `Some` but I kept `unwrap_or` if somebody reacts if we use `raw unwrapping`